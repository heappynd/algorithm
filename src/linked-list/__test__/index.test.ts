import { describe, expect, it } from 'vitest'
import { List } from '..'

describe.only('linked-list', () => {
  it('List', () => {
    const list = new List()
    expect(list).toMatchSnapshot()

    list.insertAt(0, 100)
    expect(list).toMatchSnapshot()
    list.insertAt(1, 200)
    expect(list).toMatchSnapshot()
    list.insertAt(0, 50)
    expect(list).toMatchSnapshot()

    list.removeAt(0)
    expect(list).toMatchSnapshot()

    list.removeAt(1)
    expect(list).toMatchSnapshot()
  })
})
