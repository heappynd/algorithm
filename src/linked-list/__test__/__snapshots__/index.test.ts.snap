// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`linked-list > List 1`] = `
List {
  "head": null,
  "length": 0,
}
`;

exports[`linked-list > List 2`] = `
List {
  "head": Node {
    "data": 100,
    "next": null,
  },
  "length": 1,
}
`;

exports[`linked-list > List 3`] = `
List {
  "head": Node {
    "data": 100,
    "next": Node {
      "data": 200,
      "next": null,
    },
  },
  "length": 2,
}
`;

exports[`linked-list > List 4`] = `
List {
  "head": Node {
    "data": 50,
    "next": Node {
      "data": 100,
      "next": Node {
        "data": 200,
        "next": null,
      },
    },
  },
  "length": 3,
}
`;

exports[`linked-list > List 5`] = `
List {
  "head": Node {
    "data": 100,
    "next": Node {
      "data": 200,
      "next": null,
    },
  },
  "length": 2,
}
`;

exports[`linked-list > List 6`] = `
List {
  "head": Node {
    "data": 100,
    "next": null,
  },
  "length": 1,
}
`;
